cli/src/cli_util.rs

```diff
@@ -3032,8 +3032,15 @@ fn handle_early_args(
     let early_matches = app
         .clone()
         .disable_version_flag(true)
+        // Ignore any appearance of -h/--help and continue parsing
```

Comment:
I am a bit confused about why this works. Shouldn't `.disable_help_flag().ignore_errors(true)` have been sufficient to ignore `-h` while parsing?

Reply:
In clap, help is an "error" (or the thing that triggers exceptional code path.) That's my understanding.

Reply:
yes, but shouldn't `ignore_errors` ignore this error? I guess it clearly doesn't. Maybe it's a documentation bug in clap.

Thanks again for figuring it out.

Reply:
If this is true, maybe the comment should be moved to the next line and say:

> Without this, parsing will stop on the first `-h` or `--help`, despite .disable_help_flag() and .ignore_errors().

Reply:
> yes, but shouldn't ignore_errors ignore this error, especially since the help flag is disabled? 

If the default `-h` is disabled, `-h` triggers a normal parse error. If it's enabled, `-h` triggers a "help requested error". In both cases, any remaining args won't be parsed. We do need to skip over `-h`. And that's why I put the comment there.

Reply:
Ok, that appears wrong. `.ignore_errors(true)` doesn't ignore "help error" (maybe because it isn't a parse error?) `.disable_help_flag(true)` disables the help error as well as parsing of `-h`, so we need a rule to parse `-h`.

Reply:
Updated comment.